sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/policy/TextReplacementPolicy.java

```diff
@@ -0,0 +1,215 @@
+    private final Map<String, String> textReplacementRules;
+
+    /**
+     * Creates a policy that records network calls into {@code recordedData} with replacement rules.
+     *
+     * @param recordedData The record to persist network calls into.
+     * @param textReplacementRules The replacement rules
+     */
+    public TextReplacementPolicy(RecordedData recordedData, Map<String, String> textReplacementRules) {
+        //Objects.requireNonNull(recordedData, "'recordedData' cannot be null.");
+        Objects.requireNonNull(textReplacementRules, "'textReplacementRules' cannot be null.");
+        this.recordedData = recordedData;
+        this.textReplacementRules = textReplacementRules;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        final NetworkCallRecord networkCallRecord = new NetworkCallRecord();
+        Map<String, String> headers = new HashMap<>();
+
+        captureRequestHeaders(context.getHttpRequest().getHeaders(), headers,
+            X_MS_CLIENT_REQUEST_ID,
+            CONTENT_TYPE,
+            X_MS_VERSION,
+            USER_AGENT);
+
+        networkCallRecord.setHeaders(headers);
+        networkCallRecord.setMethod(context.getHttpRequest().getHttpMethod().toString());
+
+        // Remove sensitive information such as SAS token signatures from the recording.
+        UrlBuilder urlBuilder = UrlBuilder.parse(context.getHttpRequest().getUrl());
+        if (urlBuilder.getQuery().containsKey(SIG)) {
+            urlBuilder.setQueryParameter(SIG, "REDACTED");
+        }
+        networkCallRecord.setUri(applyReplacementRule(urlBuilder.toString().replaceAll("\\?$", "")));
+
+        return next.process()
+            .doOnError(throwable -> {
+                networkCallRecord.setException(new NetworkCallError(throwable));
+                recordedData.addNetworkCall(networkCallRecord);
+                throw logger.logExceptionAsWarning(Exceptions.propagate(throwable));
+            }).flatMap(httpResponse -> {
+                final HttpResponse bufferedResponse = httpResponse.buffer();
+
+                return extractResponseData(bufferedResponse).map(responseData -> {
+                    networkCallRecord.setResponse(responseData);
+                    String body = responseData.get(BODY);
+
+                    // Remove pre-added header if this is a waiting or redirection
+                    if (body != null && body.contains("<Status>InProgress</Status>")
+                        || Integer.parseInt(responseData.get(STATUS_CODE)) == HttpURLConnection.HTTP_MOVED_TEMP) {
+                        logger.info("Waiting for a response or redirection.");
+                    } else {
+                        recordedData.addNetworkCall(networkCallRecord);
+                    }
+
+                    return bufferedResponse;
+                });
+            });
+    }
+
+    private String applyReplacementRule(String text) {
```

Comment:
Most code copied from azure-core-test with new method adopted for our requirement to replace the subscription id, tenant id, etc.